Sources/KSCrashRecording/include/KSCrashMonitorType.h

```diff
@@ -124,7 +134,7 @@ typedef enum
 /** Monitors that are required for proper operation.
  * These add essential information to the reports, but do not trigger reporting.
  */
-#define KSCrashMonitorTypeRequired (KSCrashMonitorTypeSystem | KSCrashMonitorTypeApplicationState)
+#define KSCrashMonitorTypeRequired (KSCrashMonitorTypeSystem | KSCrashMonitorTypeApplicationState | KSCrashMonitorTypeMemory)
```

Comment:
After this change the documentation is not fully correct. It's possible for `KSCrashMonitorTypeMemory` to trigger reporting. It might be overkill, but maybe it's better to split this in 2:
* A monitor that enriches reports with memory info (A)
* A monitor that tracks and reports OOMs (B)

In this case it's monitor (A) that is required while (B) is optional.

Reply:
I don't really see anyone wanting one without the other, I'd keep them in one as is, but I Leave it up to you to decide what the right way for this system is.

Reply:
It makes sense to me to have this new monitor enabled by-default as implementation.
Yet it would be really good to allow developers opt-out from sending OOM reports (e.g. if they notice too many false positives or if there's a missed edge case in the implementation).

Along with [this comment about compile-vs-runtime check](https://github.com/kstenerud/KSCrash/pull/476#discussion_r1609872510) it might make sense to have some memory-specific parameters that control how OOMs should be reported.

On top of this, it might make sense to allow developers to specify other parameters of the monitor (e.g. the memory limits) on the app side. This part is completely optional for now.

Reply:
> It makes sense to me to have this new monitor enabled by-default as implementation. Yet it would be really good to allow developers opt-out from sending OOM reports (e.g. if they notice too many false positives or if there's a missed edge case in the implementation).

Should we leave it up to the users to decide and they can do this by using a filter on the reports?

> 
> Along with [this comment about compile-vs-runtime check](https://github.com/kstenerud/KSCrash/pull/476#discussion_r1609872510) it might make sense to have some memory-specific parameters that control how OOMs should be reported.

Agreed. What other kinds of parameters do you have in mind?

Reply:
Let's have a simple `BOOL` flag for automatic OOMs reporting in `KSCrash` class (same as we have for other optional features) and enabled by default.

Talking about non-fatals, it might make sense to have an enum parameter that defines what level of available memory status should be reported as a non-fatal error. And have it disabled by defualt. To me this looks like a useful feature for some apps, but definitely not for all.

And let's see if anyone comes with request for more configurations.

Reply:
I added `ksmemory_set_nonfatal_report_level`.